maintenance/migratestatus/README.md

```diff
@@ -0,0 +1,28 @@
+# Status Context Migrator
+The migratestatus tool is a maintenance utility used to safely switch a repo from one status context to another.
+For example if there is a context named "CI tests" that needs to be moved by "CI tests v2" this tool can be used to copy every "CI tests" status into a "CI tests v2" status context and then mark every "CI tests" context as passing and retired. This ensures that no PRs are ever passing when they shouldn't be and doesn't block PRs that should be passing. The copy and retire phases can be run seperately or together at once in move mode.
+
+### Usage
+###### Modes
+This tool runs in one of three modes. Each mode is defined as a set of conditions on the statuses and an action to take if the conditions are met.
+- **copy** mode conditions on PRs that have the context to copy, but not the destination context. For each of these PRs, the 'copy' context's status is copied to the destination context.
+	- `$ ./migratestatus --copy="CI tests" --dest="CI tests v2" --tokenfile=$TOKEN_FILE --org=$ORG --repo=$REPO`
+- **retire** mode conditions on PRs that have the context to retire and the context that replaced the retired context. For each of these PRs, the context to retire has its state set to passing and its description set to "Context retired. Status moved to 'dest' context.".  This allows the retired context to be ignored and refers users to the new context.
```

Comment:
--copy only copies the state of the source context into the --dest context without modifying the source context. It does not set the source context's state to 'success' or update its description to indicate that it was retired.
Copy and retire are the two base operations, move is the combination of the two. I made copy and retire separate so that a new context can be tested after its created before retiring the original context.